src/Interpreters/Context.cpp

```diff
@@ -1813,6 +1829,56 @@ zkutil::ZooKeeperPtr Context::getZooKeeper() const
     return shared->zookeeper;
 }
 
+#if USE_HIVE
+HMSClientPtr Context::getHMSClient(const String & name) const
+{
+    using namespace apache::thrift;
+    using namespace apache::thrift::protocol;
+    using namespace apache::thrift::transport;
+    using namespace Apache::Hadoop::Hive;
+
+    std::lock_guard lock(shared->hive_metastore_mutex);
+    auto it = shared->hive_metastore_clients.find(name);
+    if (it == shared->hive_metastore_clients.end() || it->second->isExpired())
+    {
+        // connect to hive metastore
+        Poco::URI hms_url(name);
+        const auto& host = hms_url.getHost();
+        auto port = hms_url.getPort();
+
+        std::shared_ptr<TSocket> socket = std::make_shared<TSocket>(host, port);
+        socket->setKeepAlive(true);
+        socket->setConnTimeout(60000);
```

Comment:
Better to at least make 60000 defined in constexpr, it will be good lying in the same file as singleton getter class, which I mentioned earlier.

Reply:
Maybe it is ok here since `60000` is a commonly used parameter for hive metastore. It rarely need change afterwards.